### ISSUE TITLE
[Codegen]Object & UnsafeObject fails generateCodegenArtifactsFromSchema

### ISSUE THREAD

**Issue by Sunbreak** [facebook/react-native#36467]
### Description

`Object` and `UnsafeObject` is valid in flow **module** parser of `react-native-codegen`, but fails the **component** codegen process

https://github.com/facebook/react-native/blob/49dd38f1338ab8fb37144f03980bb7f122a8f04e/packages/react-native-codegen/src/parsers/flow/modules/index.js#L209-L214

### React Native Version

0.71.3

### Output of `npx react-native info`

```
info Fetching system and libraries information...
System:
    OS: macOS 12.6.1
    CPU: (8) arm64 Apple M1
    Memory: 120.03 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.19.1 - ~/.volta/tools/image/node/16.19.1/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 8.19.3 - ~/.volta/tools/image/node/16.19.1/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    Android SDK:
      API Levels: 28, 30, 31, 32, 33
      Build Tools: 30.0.2, 30.0.3, 31.0.0, 33.0.1
      System Images: android-33 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: Electric Eel 2022.1.1 Patch 2 Electric Eel 2022.1.1 Patch 2
    Xcode: 14.2/14C18 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.17 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.2.0 => 18.2.0 
    react-native: 0.71.3 => 0.71.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
info React Native v0.71.4 is now available (your project is running on v0.71.3).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.71.4
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.71.3
info For more info, check out "https://reactnative.dev/docs/upgrading".
```

### Steps to reproduce

- Initialize a library project

```sh
$ npx create-react-native-library@0.28.0 my-lib
...
✔ What type of library do you want to develop? › Fabric view
...
$ yarn install && yarn --cwd example install
```

- Add following to src

```js
// @flow

import type { HostComponent } from 'react-native/Libraries/Renderer/shims/ReactNativeTypes';
import type { UnsafeObject } from 'react-native/Libraries/Types/CodegenTypes';
import codegenNativeComponent from 'react-native/Libraries/Utilities/codegenNativeComponent';

type UnknownProps = $ReadOnly<{|
  // Error: Unknown property type for "obj": "Object" in the State
  obj: Object,
  // Error: Unknown property type for "unsafe": "UnsafeObject" in the State
  unsafe: UnsafeObject,
|}>;

type UndefinedProps = $ReadOnly<{|
  // TypeError: Cannot read properties of undefined (reading 'map')
  obj: $ReadOnly<Object>,
  // Error: Unknown property type for "unsafe": "UnsafeObject" in the State
  unsafe: $ReadOnly<UnsafeObject>,
|}>;

export default (codegenNativeComponent<UndefinedProps>(
  'MyLib',
): HostComponent<UndefinedProps>);
```

- Generate artifact from schema

```
$ cd example/android
$ ./gradlew generateCodegenArtifactsFromSchema
```

### Snack, code example, screenshot, or link to a repository

[my-lib.zip](https://github.com/facebook/react-native/files/10963117/my-lib.zip)

**Comment by cortinico** [comment]
> 0.70.6

Could you try this on 0.71?

**Comment by Sunbreak** [comment]
My bad. It was on 0.71.3

> ### React Native Version
> 0.70.6
>
> ### Output of `npx react-native info`
> ```
> info React Native v0.71.4 is now available (your project is running on v0.71.3).
> ```
